### PR TITLE
bt-287: add get current user request to mobile

### LINKED ISSUE #287: Add get current user request to mobile
Currently, when the user is signed in/up we still need to sign in again after the app relaunch. Let's add get current user request to fetch user data with stored token. If request failed we need to run logout action and user should be redirected to sign in screen

### PR DESCRIPTION
**Pull request by illia-dobro**

### CODE REVIEW

File: mobile/src/bundles/auth/screens/auth.tsx [hunk 1]
```diff
@@ -14,23 +14,22 @@ import {
...
-    const isSignUpScreen = name === AuthScreenName.SIGN_UP;
+
     const isPendingAuth = dataStatus === DataStatus.PENDING;
+
     useEffect(() => {
-        if (isSignUpScreen) {
-            void dispatch(userActions.loadAll());
-        }
-    }, [isSignUpScreen, dispatch]);
+        void dispatch(loadUser());
```
Comment (katerynakostikova) [inline mobile/src/bundles/auth/screens/auth.tsx]: It should be dispatched in root, user shouldn't see sign in screen each time he/she opens the app

File: mobile/src/bundles/auth/auth-api.ts [hunk 2]
```diff
@@ -38,6 +39,18 @@ class AuthApi extends HttpApiBase {
         return await response.json<UserSignUpResponseDto>();
     }
 
+    public async getByToken(): Promise<UserFindResponseDto> {
```
Comment (katerynakostikova) [inline mobile/src/bundles/auth/auth-api.ts]:
```suggestion
    public async getCurrentUser(): Promise<UserFindResponseDto> {
```

File: mobile/src/bundles/auth/store/actions.ts [hunk 3]
```diff
@@ -30,11 +31,20 @@ const signUp = createAsyncThunk<
     }
 });
 
+const loadUser = createAsyncThunk<
```
Comment (katerynakostikova) [inline mobile/src/bundles/auth/store/actions.ts]:
```suggestion
const loadCurrentUser = createAsyncThunk<
```

File: mobile/src/navigations/root/root.tsx [hunk 4]
```diff
@@ -24,6 +30,21 @@ const screenOptions: NativeStackNavigationOptions = {
...
+
+    const isPendingAuth = dataStatus === DataStatus.PENDING;
+
+    useEffect(() => {
+        void dispatch(loadUser());
+    }, [dispatch]);
+
+    if (isPendingAuth) {
+        return <Overlay isActive={isPendingAuth} />;
+    }
```
Comment (katerynakostikova) [inline mobile/src/navigations/root/root.tsx]:
```suggestion
    if (isPendingAuth) {
        return null
    }
```

**Review by katerynakostikova — APPROVED** [review]

**Review by anogire — APPROVED** [review]

**Pull request merged**